### PR TITLE
Process of adding Whaley's Locations (9 of 9)

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
**[1 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25070)** - Combines/Renames/Moves/Lints mapgen files most notably in /mapgen/cosmics_additions.

**[2 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25072)** - Combines/Renames/Moves/Lints mapgen files most notably in /mapgen/parks_recreation & /mapgen/small_town_buildings.

**[3 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24924)** - Adds Whaley's official (in-game) mod, "More Buildings" to the vanilla game.

**[4 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24989)** - Adds Whaley's Zoo (from his "Big Buildings mod).

**[5 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24990)** - Adds Whaley's Stadium (from his "Big Buildings" mod).

**[6 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24992)** - Adds the remainder of Whaley's "Big Buildings" mod.

**[7 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24998)** - Adds Whaley's "City Entertainment" mod.

**[8 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25000)** - Adds Whaley's "Seedy Spots" mod.

---

**9 of 9** - Adds Whaley's "River Buildings" mod.

- Adds boat rentals, dirt roads, and moonshine camps to mapgen folder.
- Adds rural houses and river houses to house foler.
- Adds crazy person npc.
- Adds boats, boat vehicle group and vehicle parts.
- Adds overmap_terrain and overmap_special info.

**This PR is NOT dependent on any other step.**

Discussion for the addition of Whaley's Mods can be found **[here](https://github.com/CleverRaven/Cataclysm-DDA/issues/24932)**.

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
Not sure why gorgon is failing on this one...

**Comment by mlangsdorf** [comment]
Please don't mainline boats at this time. Put all this stuff in data/mods/Boats where it belongs for now.

**Comment by cosmiccoincidence** [comment]
Moved boats, related items, related locations and other related info to "Boats" mod.

**Comment by cosmiccoincidence** [comment]
Wait that last commit didn't add the files to the "Boats" mod folder.
This happened last time I added changes inside the mod folder, github desktop didn't recognize the changes...

**Comment by cosmiccoincidence** [comment]
Had to upload the files via github website, but it worked so..

**Comment by ZhilkinSerg** [comment]
Had to update some comments (move them to objects) as they cause game loading errors.

File: data/mods/Boats/whaleys_boats.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    ]
+  },
+  {
+    "//": "Place the following in vehicle_groups.json in /data/json/",
+    "type": "vehicle_group",
+    "id": "boatrent",
+    "vehicles": [ [ "canoe", 2000 ], [ "raft", 2000 ] ]
+  },
+  {
+    "//": "Place the following in recipe_vehicle.json in /data/json/recipes/",
```
Comment (Brambor) [inline data/mods/Boats/whaleys_boats.json]: Isn't something supposed to be done with this?

Comment (ZhilkinSerg) [reply data/mods/Boats/whaleys_boats.json]: Yes, but later when boat vehicles would go to mainline from mod.

**Pull request merged**